src/OmniSharp.Abstractions/Endpoints/IGotoDefinition.cs

```diff
@@ -0,0 +1,11 @@
+using System.Threading.Tasks;
+using OmniSharp.Models;
+
+namespace OmniSharp.Endpoints
+{
+    public interface IGotoDefintion
+    {
+        Task<bool> IsApplicableTo(GotoDefinitionRequest request);
```

Comment:
So we could, but should we?  I suppose it's more `Functional`, lol.

We could do something like...

    public static class OmniSharpEndpoints
    {
        public static Type GotoDefinition {get;} = typeof(Func<GotoDefinitionRequest, Task<GotoDefintionResponse>>)
    }

    class RoslynServices
    {
        [OmniSharpEndpoint(OmniSharpEndpoints.GotoDefinition, Languages.CSharp)]
        public async Task<GotoDefinitionResponse> GotoDefinition(GotoDefinitionRequest request) { ... }
    }

That just feels verbose on the C# side, but may be easier on the F# side of things?  Could we support both?  And additionally have....

    [OmniSharpEndpoint(typeof(IGotoDefinition), Languages.CSharp")]
    class RoslynServices : IGotoDefinition { ... }

Then there is just an extra check to find the correct service, for the given language (which could be thrown behind a cache).

I'm not against the former, it would enforce some extra rules on our part to make sure we
a) never change the delegates
b) make sure we document any changes to the delegates up front.

Note: `OmniSharpEndpointAttribute` would derive from `ExportAttribute` and implement metadata attribute.  On the import side, we would import that metadata with the language.